### ISSUE TITLE
Simplify Noise2Void

### ISSUE THREAD

**Issue by jdeschamps** [CAREamics/careamics#307]
## Description

`N2VManipulate` as a transform makes everything complicated, including a lot of constraints on the Pydantic models and the potential for users to miss-configure their training, in particular in the Lightning API.

In addition, this transform returns a different number of objects has it returns also the original patch. The reason for that is the current implementation of the loss.

Overall, N2V requires many additional checks throughout the code.

### Which part of the code?

Examples of N2V checks:
https://github.com/CAREamics/careamics/blob/cd2eae26252b8ef0618b5b7e6fcdac3f5a58d8f9/src/careamics/config/data_model.py#L238

N2V loss:
https://github.com/CAREamics/careamics/blob/cd2eae26252b8ef0618b5b7e6fcdac3f5a58d8f9/src/careamics/losses/fcn/losses.py#L53

In dataset:
https://github.com/CAREamics/careamics/blob/4d4688a2498364b1d7c6cca6758c34047be26100/src/careamics/dataset/in_memory_dataset.py#L222

### Simplify transform

In `N2VManipulate`:
``` python
return patch * mask
```

### Simplify loss

In `n2v_loss`, simply pass `target`:

``` python
mask = (target != 0).float()
errors = (target - manipulated_patches) ** 2
loss = torch.sum(errors * mask) / torch.sum(mask)
```

### `N2VModule` as a subclass of `UNetModule`

`N2VModule` as a subclass of `UNetModule` would allow injecting the `N2VManipulate` transform. We would need to see how to instantiate it properly based on the configuration.

This would also simplify using N2V in the Lightning API, which is one way users have told us they intend to use N2V.

**Comment by jdeschamps** [comment]
In the current code base, we get the original patch in the Lightning module and therefore there is no need to return it from the N2VManipulate.

**Comment by jdeschamps** [comment]
Outdated but the simplification of N2V manipulate for loss calculations could be inspected futher.
